sotodlib/coords/optics.py

```diff
@@ -519,32 +510,36 @@ def SAT_focal_plane(aman, x=None, y=None, pol=None, roll=0, mapping_data=None):
         pol = aman.focal_plane.pol_fp
 
     if mapping_data is None:
-        fp_to_sky = sat_to_sky(SAT_X, SAT_LON)
+        fp_to_sky = sat_to_sky(SAT_R_FP, SAT_R_SKY)
     else:
         mapping_data = (tuple(val) for val in mapping_data)
         fp_to_sky = sat_to_sky(*mapping_data)
 
-    # NOTE: lonlat coords are naturally centered at (1, 0, 0) and
-    #       xieta at (0, 0, 1). The euler angle below does this recentering
-    #       as well as flipping the sign of eta.
-    #       There is also a sign flip of xi that is supresses the factor of
-    #       -1 that would normally be applied when calculating lon since
-    #       it has the opposite sign as x.
-    #       The sign flips perform the flip about the origin from optics.
-    minus_lon = np.sign(x) * fp_to_sky(np.abs(x))
-    lat = np.sign(y) * fp_to_sky(np.abs(y))
-    _xi, _eta, _ = quat.decompose_xieta(
-        quat.euler(1, np.deg2rad(90)) * quat.rotation_lonlat(minus_lon, lat)
-    )
+    # Get things in polor coords 
+    r_fp = (x**2 + y**2)**.5
+    phi_fp = np.arctan2(y, x)
+
+    # Now to the sky
+    theta_iso = -fp_to_sky(r_fp)
+    phi_iso = np.pi/2 - phi_fp
+
+    # Flip about the origin for optics (180 deg rotation)
+    phi_iso += np.pi
+
+    # Now go to xieta
+    _xi, _eta, _ = quat.decompose_xieta(quat.rotation_iso(theta_iso, phi_iso))
+
+    # Apply roll
     xi = _xi * np.cos(np.deg2rad(roll)) - _eta * np.sin(np.deg2rad(roll))
     eta = _eta * np.cos(np.deg2rad(roll)) + _xi * np.sin(np.deg2rad(roll))
 
+    # Lets do gamma as a set of endpoints
     pol_x, pol_y = gen_pol_endpoints(x, y, pol)
-    pol_minus_lon = np.sign(pol_x) * fp_to_sky(np.abs(pol_x))
-    pol_lat = np.sign(pol_y) * fp_to_sky(np.abs(pol_y))
-    _xi, _eta, _ = quat.decompose_xieta(
-        quat.euler(1, np.deg2rad(90)) * quat.rotation_lonlat(pol_minus_lon, pol_lat)
-    )
+    pol_r = (pol_x**2 + pol_y**2)**.5
+    pol_phi = np.arctan2(pol_y, pol_x)
+    pol_theta_iso = -fp_to_sky(pol_r)
+    pol_phi_iso = np.pi/2 - pol_phi + np.pi
+    _xi, _eta, _ = quat.decompose_xieta(quat.rotation_iso(pol_theta_iso, pol_phi_iso))
     pol_xi = _xi * np.cos(np.deg2rad(roll)) - _eta * np.sin(np.deg2rad(roll))
     pol_eta = _eta * np.cos(np.deg2rad(roll)) + _xi * np.sin(np.deg2rad(roll))
     gamma = get_gamma(pol_xi, pol_eta)

```

Comment:
I see you are insisting on computing some kind of projected angle here, which I have my concerns about.  Ok.

But if we're going to do it this way, I at least need it to not be done in a flat sky approximation.  The present approach uses delta xi and delta eta to measure an angle.  But the local xi and eta axes are not orthogonal on the sky.  The spherical way would be to take your mini-vectors, and rotate them back to near z^ before computing the angle.

To do that, I believe you can just modify the get_gamma function:
```
def get_gamma(pol_xi, pol_eta):
    # docstring etc

    xi = pol_xi.reshape((-1, 2))
    eta = pol_eta.reshape((-1, 2))

    q0 = quat.rotation_xieta(xi[:,0], eta[:,0])
    q1 = quat.rotation_xieta(xi[:,1], eta[:,1])
    dq = ~q0 * q1
    d_xi, d_eta, _ = quat.decompose_xieta(dq)

    gamma = np.arctan2(d_xi, d_eta) % (2 * np.pi)
    return gamma
```